substrate/frame/multisig/src/lib.rs

```diff
@@ -481,16 +516,122 @@ pub mod pallet {
+		/// not the first approval, then it must be `Some`, with the timepoint (block number and
+		/// transaction index) of the first approval transaction.
+		/// - `call_hash`: The hash of the call to be executed.
+		/// - `maybe_expiry`: An optional expiry of the multisig operation defined in block number.
+		///
+		/// ## Complexity
+		/// - `O(S)`.
+		/// - Up to one balance-reserve or unreserve operation.
+		/// - At most two inserts, both `O(S)` where `S` is the number of signatories. 
+		///   `S` is capped by `MaxSignatories`, with weight being proportional.
+		/// - One encode & hash, both of complexity `O(S)`.
+		/// - Up to one binary search and insert (`O(logS + S)`).
+		/// - I/O: 2 reads `O(S)`, up to 2 mutations `O(S)`. Up to two removes.
+		/// - One event.
+		/// - Storage: inserts up to two items. One of them is bounded by `MaxSignatories`, with a 
+		///   deposit taken for its lifetime of `DepositBase + threshold * DepositFactor`. The other
+		///   value is a `BlockNumber`.
+		#[pallet::call_index(4)]
+		#[pallet::weight({
+			let s = other_signatories.len() as u32;
+
+			T::WeightInfo::approve_as_multi_create(s)
+				.max(T::WeightInfo::approve_as_multi_approve(s))
+				.saturating_add(*max_weight)
+		})]
+		pub fn approve_as_multi(
+			origin: OriginFor<T>,
+			threshold: u16,
+			other_signatories: Vec<T::AccountId>,
+			maybe_timepoint: Option<Timepoint<BlockNumberFor<T>>>,
+			call_hash: [u8; 32],
+			max_weight: Weight,
+			maybe_expiry: Option<BlockNumberFor<T>>,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			Self::operate(
+				who,
+				threshold,
+				other_signatories,
+				maybe_timepoint,
+				CallOrHash::Hash(call_hash),
+				max_weight,
+				maybe_expiry,
+			)
+		}
+
+		/// Clear up all the state of an expired multisig operation. Any deposit reserved previously
+		/// for this operation will be unreserved on success.
+		///
+		/// Anyone can call this and free up the not needed state of the expired multisig.
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// - `threshold`: The total number of approvals for this dispatch before it is executed.
+		/// - `signatories`: The accounts who can approve this dispatch.
+		/// - `timepoint`: The timepoint (block number and transaction index) of the first approval
+		/// transaction for this dispatch.
+		/// - `call_hash`: The hash of the call to be executed.
+		///
+		/// ## Complexity
+		/// - `O(S)`.
+		/// - Up to one balance-reserve or unreserve operation.
+		/// - At most two inserts, both `O(S)` where `S` is the number of signatories. 
+		///   `S` is capped by `MaxSignatories`, with weight being proportional.
+		/// - One encode & hash, both of complexity `O(S)`.
+		/// - One event.
+		/// - I/O: 2 reads `O(S)`, up to two removes.
+		/// - Storage: removes up to two items.
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::WeightInfo::cancel_as_multi(signatories.len() as u32))]
+		pub fn clear_expired_multi(
```

Comment:
Should add to the doc comment at the top of the file, under `Dispatchable functions`